### DISCUSSION TITLE
Typehints and typerrors for "run" method

### DISCUSSION THREAD

**Discussion by msnegurski** [lorisleiva/laravel-actions#255] [Ideas]
Hey, would be great if "run" method provided typehints and errors that are same as in "handle" method.

For example, for "handle" function `public function handle(string $query): array` would be great if "run" method required `string` input and had return type of `array`.

I don't know how feasable it is though. Tried to implement it in my project but with no success - PHP desparately lacks generics.

**Comment by pindab0ter** [comment]
This is provided by the [Laravel Idea plugin](https://laravel-idea.com/) for anyone using PHPStorm.

**Comment by Wulfheart** [comment]
There is also an IDE helper package. I have currently no idea how we could implement that better in this package.
